### PR TITLE
General refactoring: `nim-waku`

### PR DESCRIPTION
**Pull request by jm-clius**
This PR refactors `nim-waku` according to [Track 1 refactoring checklist](https://hackmd.io/1imOGULZRsed2HpgmzGleA?view)

It covers all modules not addressed in the existing/closed PRs: https://github.com/status-im/nim-waku/pull/664 #667.
Not everything on the Track 1 checklist is necessary for modules in `/test`-directory, so I've mostly focused on `import` cleanup.

Once all Track 1 refactoring PRs are merged, another pass of code maintenance (especially `import` cleanups) will be necessary to account for the effect of new `export`s and other improvements.

### CODE REVIEW

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -156,16 +154,23 @@ proc printReceivedMessage(c: Chat, msg: WakuMessage) {.raises: [Defect].} =
     trace "Printing message", topic=DefaultTopic, chatLine,
       contentTopic = msg.contentTopic
 
-proc selectRandomNode(fleetStr: string): string =
+proc selectRandomNode(fleetStr: string): Option[string] =
```
Comment (oskarth) [inline examples/v2/chat2.nim:157]: If a select operation fails, should it not be a Result type that can error?

Alternatively, if we are getting it from a list and the list happens to be empty, then Option makes sense.

Here it seems like the empty case is a HTTP exception, which seems more like an error to me. Semantics though...

Comment (oskarth) [reply examples/v2/chat2.nim:157]: With error the exception message can be propagated upward too, such as if it is a permission issue, server down, change of format for fleets json, etc.

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -295,6 +300,7 @@ proc readInput(wfd: AsyncFD) {.thread.} =
     let line = stdin.readLine()
     discard waitFor transp.write(line & "\r\n")
 
+{.pop.} # @TODO confutils.nim(775, 17) Error: can raise an unlisted exception: ref IOError
```
Comment (oskarth) [inline examples/v2/chat2.nim:303]: Oh til

**Review by oskarth — APPROVED** [review]

File: examples/v2/chat2.nim (hunk 1 above)
Comment (jm-clius) [reply examples/v2/chat2.nim:157]: Agree. `Result` seems more idiomatic here.

File: examples/v2/chat2.nim (hunk 2 above)
Comment (jm-clius) [reply examples/v2/chat2.nim:303]: Yeah, it's a life-saver, as it's a pain to get around the `confutils` issue otherwise (read: I don't know how :D)

**Pull request merged**